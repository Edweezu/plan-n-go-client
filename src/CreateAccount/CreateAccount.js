import React from 'react'
import { Link } from 'react-router-dom'
import UsersApiService from '../services/users-api-service'

export default class CreateAccount extends React.Component {

    constructor (props) {
        super (props)
        this.state = {
            error: null
        }
    }

    handleSubmit = (e) => {
        e.preventDefault()
        const { username, password } = e.target
        const newUser = {
            username: username.value,
            password: password.value
        }

        return UsersApiService.postUser (newUser)
            .then(responseJson => {
                username.value = ''
                password.value = ''
                this.handleRegistration()
                console.log('user created')
            })
            .catch(responseJson => {
                this.setState({
                    error: responseJson.error
                })
            })

    }

    handleRegistration = () => {
        const { history } = this.props
        history.push('/login')
    }



    render () {
        const { error } = this.state

        return (
            <main className='RegisterPage'>
                <h2>
                    Create an Account
                </h2>
                <form className='login-form' onSubmit={this.handleSubmit}>
                    <div role='alert'>
                        {error && <p className='red'>{error}</p>}
                    </div>
                    <div className='signup-element'>
                        <div className='signup-label'>
                            <label htmlFor='username'>
                                Username    
                            </label>
                            <span className='astrik'>
                                *
                            </span>
                        </div>
                        <div className='signup-input'>
                            <input id='username' name='username' type='text'
                            required/>
                        </div>
                    </div>
                    <div className='signup-element'>
                        <div className='signup-label'> 
                            <label htmlFor='password'>
                                Password    
                            </label>
                            <span className='astrik'>*</span>
                        </div>
                        <div className='signup-input'>
                            <input id='password' name='password' type='password'
                            required/>
                        </div>
                    </div>
                    <div className='signup-element'>
                        <div className='signup-label'>
                            <label htmlFor='confirm-pw'>
                                Confirm Password    
                            </label>
                            <span className='astrik'>*</span>
                        </div>
                        <div className='signup-input'>
                            <input id='confirm-pw' name='confirm-pw' type='password' required/>
                        </div>
                    </div>
                    <div className='signin-button'>
                        <button className="btn btn-sign-in" type='submit'>
                            Create an Account
                        </button>
                    </div>
                    <div className='login-form-redirect'>
                        <Link to='/login'>Already have an Account?</Link>
                    </div>
                </form>
            </main>
            
        )
    }  
}