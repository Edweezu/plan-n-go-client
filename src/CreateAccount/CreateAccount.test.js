import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import CreateAccount from './CreateAccount'


describe('CreateAccount component', () => {
    it('renders the given component', () => {
        const wrapper = shallow(<CreateAccount />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})