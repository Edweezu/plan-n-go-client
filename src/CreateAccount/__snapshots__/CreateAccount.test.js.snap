// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateAccount component renders the given component 1`] = `
<main
  className="RegisterPage"
>
  <h2>
    Create an Account
  </h2>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="signup-element"
    >
      <div
        className="signup-label"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="signup-input"
      >
        <input
          id="username"
          name="username"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="signup-element"
    >
      <div
        className="signup-label"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="signup-input"
      >
        <input
          id="password"
          name="password"
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="signup-element"
    >
      <div
        className="signup-label"
      >
        <label
          htmlFor="confirm-pw"
        >
          Confirm Password
        </label>
        <span
          className="astrik"
        >
          *
        </span>
      </div>
      <div
        className="signup-input"
      >
        <input
          id="confirm-pw"
          name="confirm-pw"
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="signin-button"
    >
      <button
        className="btn btn-sign-in"
        type="submit"
      >
        Create an Account
      </button>
    </div>
    <div
      className="signup-demo"
    >
      <p>
        Password must be longer than 8 characters and contain one upper case, lower case, number and special character.
      </p>
    </div>
    <div
      className="login-form-redirect"
    >
      <Link
        to="/login"
      >
        Already have an Account?
      </Link>
    </div>
  </form>
</main>
`;
